controllers/kuadrant_controller.go

```diff
@@ -469,27 +471,69 @@ func (r *KuadrantReconciler) removeAnnotationFromGateways(ctx context.Context, k
 }
 
 func (r *KuadrantReconciler) reconcileLimitador(ctx context.Context, kObj *kuadrantv1beta1.Kuadrant) error {
-	limitador := &limitadorv1alpha1.Limitador{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Limitador",
-			APIVersion: "limitador.kuadrant.io/v1alpha1",
-		},
+	limitadorKey := client.ObjectKey{Name: common.LimitadorName, Namespace: kObj.Namespace}
+	limitador := &limitadorv1alpha1.Limitador{}
+	err := r.Client().Get(ctx, limitadorKey, limitador)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			limitador = &limitadorv1alpha1.Limitador{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Limitador",
+					APIVersion: "limitador.kuadrant.io/v1alpha1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      common.LimitadorName,
+					Namespace: kObj.Namespace,
+				},
+				Spec: limitadorv1alpha1.LimitadorSpec{
+					RateLimitHeaders: &[]limitadorv1alpha1.RateLimitHeadersType{limitadorv1alpha1.RateLimitHeadersTypeDraft03}[0],
+					Telemetry:        &[]limitadorv1alpha1.Telemetry{limitadorv1alpha1.TelemetryExhaustive}[0],
+				},
+			}
+		} else {
+			return err
+		}
+	}
+
+	if kObj.Spec.Limitador != nil {
+		if kObj.Spec.Limitador.Affinity != nil {
+			limitador.Spec.Affinity = kObj.Spec.Limitador.Affinity
+		}
+		if kObj.Spec.Limitador.PodDisruptionBudget != nil {
+			limitador.Spec.PodDisruptionBudget = kObj.Spec.Limitador.PodDisruptionBudget
+		}
+		if kObj.Spec.Limitador.Replicas != nil {
+			limitador.Spec.Replicas = kObj.Spec.Limitador.Replicas
+		}
+		if kObj.Spec.Limitador.ResourceRequirements != nil {
+			limitador.Spec.ResourceRequirements = kObj.Spec.Limitador.ResourceRequirements
+		}
+		if kObj.Spec.Limitador.Storage != nil {
+			limitador.Spec.Storage = kObj.Spec.Limitador.Storage
+		}
+	}
+
+	tmp := &limitadorv1alpha1.Limitador{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      common.LimitadorName,
+			Name:      "temp",
 			Namespace: kObj.Namespace,
 		},
-		Spec: limitadorv1alpha1.LimitadorSpec{
-			RateLimitHeaders: &[]limitadorv1alpha1.RateLimitHeadersType{limitadorv1alpha1.RateLimitHeadersTypeDraft03}[0],
-			Telemetry:        &[]limitadorv1alpha1.Telemetry{limitadorv1alpha1.TelemetryExhaustive}[0],
-		},
+	}
+	err = r.SetOwnerReference(kObj, tmp)
+	if err != nil {
+		return err
+	}
+
+	if limitador.OwnerReferences != nil && !reflect.DeepEqual(tmp.OwnerReferences, limitador.OwnerReferences) {
+		return nil
 	}
 
-	err := r.SetOwnerReference(kObj, limitador)
+	err = r.SetOwnerReference(kObj, limitador)
```

Comment:
There is a bug here in my mutate function, working on the fix now, but should explain what is going on here. 

Firstly the Kuadrant operator expects the limitador CR to be called `limitador` and in the same namespace as the Kuadrant CR. The code will not reconcile any limitador resources of a different name or in different namespace. 

The `reflect.DeepEqual` on line 527 is me being extra safe. I would only expect that to true if the user has its own custom controller that is managing the limitador CR. I could see this being a thing if the user was to use limitador as standalone product before using kuadrant. 

So the bug is in the setting of the owner reference if the user manually creates the limitador CR in the same namespace as the kuadrant CR when the kuadrant operator is scaled down for example. Or the real world use case would be if the user was using the limitador as a standalone product and then starts to use kuadrant. In this case you would expect kuadrant to take ownership of the limitador CR.

Reply:
Yeah, this is an edge case. I guess to me, if I were to use limitador standalone first (using the `limitador` name) in the kuadrant namespace, having kuadrant take ownership of the CR afterwards may not necessary be expected, as this limitador could be used by other services 🤔 Deleting the kuadrant CR, will then delete this limitador instance, potentially affecting other services using this limitador instance 🤔 

Anyhow, just a thought, I don't mind as long as this is expected flow, though maybe best documented somewhere that this will happen at least.

In the case where the limitador CR is managed by another custom controller (due to different owner references), I think a log or a even better a status on the kuadrant CR noting this would be nice, since it is not obvious to the user that the limitador spec defined in the Kuadrant CR will not be reconciled due to this

Reply:
I agree with Kevin here, and to add a bit more, it could be a common scenario to have a _limitador_ instance in the cluster with its default name `limitador`, thus Kuadrant should not ownership if not explicitly declared in the Kuadrant CR.

Reply:
I agree it could be common to have any existing limitador CRs called `limitador` on the cluster. But let me ask this question. If you have a limitador instance that you do not want kuadrant to manage, why would you create the kuadrant CR in the same namespace as the existing limitador CR? Next question I would ask, can limitador work with two limitador CRs in the same namespace?